cranelift/reader/src/lib.rs

```diff
@@ -72,19 +72,17 @@ pub fn parse_sets_and_triple(flag_set: &[String], flag_triple: &str) -> Result<O
     // Collect unknown system-wide settings, so we can try to parse them as target specific
     // settings, if a target is defined.
     let mut unknown_settings = Vec::new();
-    match parse_options(
-        flag_set.iter().map(|x| x.as_str()),
-        &mut flag_builder,
-        Location { line_number: 0 },
-    ) {
-        Err(ParseOptionError::UnknownFlag { name, .. }) => {
-            unknown_settings.push(name);
-        }
-        Err(ParseOptionError::UnknownValue { name, value, .. }) => {
-            unknown_settings.push(format!("{}={}", name, value));
+    for flag in flag_set {
```

Comment:
Note that this bug fix isn't directly related to this PR but it affected one test. Specifically the bug was that if there were two target-specific options the second one would be discarded because `parse_options` would only return one error for the first one it didn't recognize for the top-level flags. Now this unconditionally parses all flags which is, I believe, the intended behavior.